internal/exec/exec.go

```diff
@@ -55,29 +91,48 @@ func executeCommand(cmd *exec.Cmd) error {
 	// Start indicator
 	shutdownCh := make(chan struct{})
 	go indicator(shutdownCh)
+	defer close(shutdownCh) // Signal indicator() to terminate
 
-	// Execute the process immediately
+	// Start the process without waiting for it to finish
 	if err = cmd.Start(); err != nil {
 		return errors.Wrapf(err, execFailedOnStartMsg, cmd.Path)
 	}
-	// Stream command output:
-	// Creates a bufio.Scanner that will read from the pipe
-	// that supplies the output written by the process.
-	scanout, scanerr := scanner(stdout, stderr)
 
-	if scanerr.Err() != nil {
-		return errors.Wrapf(err, execFailedOnScanMsg, cmd.Path)
-	}
+	// Wait for the process to finish in a goroutine. We wait until it finishes or termination is requested via terminateCh.
+	finishedCh := make(chan error, 1)
+	go func() {
+		// Stream command output:
+		// Creates a bufio.Scanner that will read from the pipe
+		// that supplies the output written by the process.
+		// Note: this waits until the process finishes or an error occurs.
+		scanout, scanerr := scanner(stdout, stderr)
+
+		if scanerr.Err() != nil {
+			finishedCh <- errors.Wrapf(err, execFailedOnScanMsg, cmd.Path)
+			return
+		}
 
-	if scanout.Err() != nil {
-		return errors.Wrapf(err, execFailedOnErrorGetMsg, cmd.Path)
-	}
+		if scanout.Err() != nil {
+			finishedCh <- errors.Wrapf(err, execFailedOnErrorGetMsg, cmd.Path)
+			return
+		}
 
-	// Get execution success or failure:
-	if err = cmd.Wait(); err != nil {
-		return errors.Wrapf(err, execFailedOnFinishMsg, cmd.Path)
+		// Get execution success or failure
+		finishedCh <- cmd.Wait()
+	}()
+
+	select {
+	case err = <-finishedCh:
+		if err != nil {
+			return errors.Wrapf(err, execFailedOnFinishMsg, cmd.Path)
+		}
+	case <-terminateCh:
+		// Kill the process. We don't care if an error occurs here, we did our best and it doesn't affect the user.
+		_ = cmd.Process.Kill()
+		// Return an error so that we don't continue to the next process
+		return fmt.Errorf(execKilledMsg)
```

Comment:
please use errors.New

Reply:
Not sure if this is necessary since we don't do anything with this error so we don't need the stack trace